ARCs/arc-0018.md

```diff
@@ -0,0 +1,601 @@
+
+> A transfer matching the offer would require the transfer amount <= offered amount and that the transfer is sent by auth_address
+
+#### Royalty Asset Parameters 
+
+The Clawback parameter **MUST** be set to the Application Address of the Royalty Enforcer.
+
+> Since the Royalty Enforcer relies on using the Clawback mechanism to perform the transfer the Clawback should NEVER be set to the zero address. 
+
+The Freeze parameter **MUST** be set to the Application Address of the Royalty Enforcer if `FreezeAddr != ZeroAddress`, else set to `ZeroAddress`.
+
+> If the creator wants to allow an ASA to be Royalty Free after some conditions are met, it should be set to the Application Address 
+
+The Manager parameter **MUST** be set to the Application Address of the Royalty Enforcer if `ManagerAddr != ZeroAddress`, else set to `ZeroAddress`.
+
+> If the creator wants to update the Freeze parameter, this should be set to the application address
+
+The Reserve parameter **MAY** be set to anything.
+
+The `DefaultFrozen` **MUST** be set to true.
+
+### Third Party Marketplace
+
+In order to support secondary sales on external markets this spec was designed such that the Royalty Asset may be listed without transferring it from the current owners account.  
+
+A Marketplace may call the transfer request as long as the address initiating the transfer has been set as the `authorized_address` through the [offer](#offer) method in some previous transaction by the current owner.
+
+### ABI Methods
+
+The following is a set of methods that conform to the [ABI](arc-0004.md) specification meant to enable the configuration of a Royalty Policy and perform transfers.
+
+#### Set Policy  (REQUIRED):
+    
+```
+set_policy(royalty_basis: uint64, royalty_recipients: account)
+```
+
+Sets the policy for any assets using this application as a Royalty Enforcer.
+
+The `royalty_basis` is the percentage for royalty payment collection, specified in basis points (e.g., 1% is 100).  
+
+A Royalty Basis **MUST** be immutable, if an application call is made that would overwrite an existing value it **MUST** fail. 
+
+> The immutability of a royalty basis is important since mutability introduces the possibility for a situation where, after an initial sale, the royalty policy is updated from 1% to 100% for example. This would make any further sales have the full payment amount sent to the royalty recipient and the seller would receive nothing.
+
+The `royalty_receiver` is the address of the account that should receive a partial share of the payment for any [transfer](#transfer) of an asset subject to royalty collection.
+
+The Royalty Collector **MUST** be opted into the full set of assets contained in this list of payment_assets. 
+
+> In the case that an account is not opted into an asset, any transfers where payment is specified for that asset will fail until the account opts into the asset. or the policy is updated.
+
+
+#### Set Payment Asset (REQUIRED):
```

Comment:
Why are we making this required? If they want to only accept algos

Reply:
Only required to be implemented, not required to be used